### PR TITLE
[TBBAS-1459] Python bindings: MultiReader

### PR DESCRIPTION
**Pull request by Yarikk26**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description:

- **MultiReader** and **TimeMultiReader** added to Python bindings
- Tests updated
- **MockSignal** updated

### CODE REVIEW

File: bindings/python/core_types/include/py_core_types/py_converter.h [hunk 1]
```diff
@@ -35,14 +36,21 @@ inline py::array_t<typename Sequence::value_type> toPyArray(Sequence&& seq, cons
...
 
     py::array::ShapeContainer arrayShape = {size};
-    if(!shape->empty()) arrayShape = std::move(shape);
+    if (!shape->empty())
+        arrayShape = std::move(shape);
+
+    if (strides->empty())
+    {
+        return py::array(dt, arrayShape, data, capsule);
+    }
```
Comment (mojca) [inline bindings/python/core_types/include/py_core_types/py_converter.h:53]: Just nitpicking. I would say that it's better to have the code consistent: either `{}` everywhere for single-line statements, or nowhere (two lines above it's without).
```suggestion
        return py::array(dt, arrayShape, data, capsule);
```

File: bindings/python/opendaq/include/py_opendaq/py_typed_reader.h [hunk 2]
```diff
@@ -179,82 +181,129 @@ struct PyTypedReader
...
+
         if constexpr (std::is_same_v<ReaderType, daq::BlockReaderPtr>)
         {
             reader->getBlockSize(&blockSize);
         }
+        if constexpr (isMultiReader)
+        {
+            daq::ReaderConfigPtr readerConfig = reader.template asPtr<daq::IReaderConfig>();
+            blockSize = readerConfig.getInputPorts().getCount();
+        }
```
Comment (JakaMohorkoDS) [inline bindings/python/opendaq/include/py_opendaq/py_typed_reader.h:195]: General question - can these not be cached on construction?

**Review by JakaMohorkoDS — CHANGES_REQUESTED** [review]
Lgtm, but should also include the multi reader builder wrapper.

File: bindings/python/opendaq/include/py_opendaq/py_typed_reader.h (hunk 2 above)
Comment (Yarikk26) [reply bindings/python/opendaq/include/py_opendaq/py_typed_reader.h:195]: What do you mean?

**Comment by adolenc** [comment]
> Lgtm, but should also include the multi reader builder wrapper.

Aren't openDAQ builders just workarounds for C++ functions not supporting calls with an arbitrary combination of (keyword) arguments? For python the `**kwargs` on creation functions are the pythonic version of builders, if anything I would say using builders in python is an antipattern.

**Review by JakaMohorkoDS — APPROVED** [review]

**Pull request merged**